src/NServiceBus.Metrics/Core/Snapshotting.cs

```diff
@@ -0,0 +1,84 @@
+﻿namespace NServiceBus.Metrics
+{
+    using System;
+    using System.Linq;
+    using global::Metrics.MetricData;
+
+    static class MetricsDataExtentions
+    {
+
+        public static MetricsData SnapshotMetrics(this MetricsData data, DateTime reportWindowStart, DateTime reportWindowEnd)
+        {
+            var environment = data.Environment.ToList();
+
+            environment.Add(new EnvironmentEntry("ReportWindowStart", $"{reportWindowStart:yyyy-MM-ddTHH:mm:ss.FFFZ}"));
+            environment.Add(new EnvironmentEntry("ReportWindowEnd", $"{reportWindowEnd:yyyy-MM-ddTHH:mm:ss.FFFZ}"));
+
+            return new MetricsData(
+                data.Context,
+                DateTime.UtcNow,
+                environment,
+                data.Gauges.Select(g => g.Snapshot()).ToArray(),
+                data.Counters.Select(c => c.Snapshot()).ToArray(),
+                data.Meters.Select(m => m.Snapshot(true)).ToArray(),
+                data.Histograms.Select(h => h.Snapshot(true)).ToArray(),
+                data.Timers.Select(t => t.Snapshot(true)).ToArray(),
+                data.ChildMetrics);
+        }
+    }
+
+    static class GagueValueSourceExtentions
```

Comment:
just dump all these in one extension class. no point in having a class per method